Chess/Assets/Player.cs

```diff
@@ -206,18 +242,19 @@ private Task SubscribeToPlayerMessages()
 
     public void PlayerInteract(InputAction.CallbackContext context)
     {
-        if (!context.performed && _currentLobby != null) return;
+        if (!context.performed && _currentSession != null) return;
         var mousePosition = Mouse.current.position.ReadValue();
         var rayOrigin = playerCamera.ScreenPointToRay(mousePosition);
         if (Physics.Raycast(rayOrigin, out var hitInfo))
         {
-            // TODO: check if a piece is selected and another piece is clicked - is that a move?
-            if (hitInfo.transform.gameObject.name == "Board")
+            var gameObject = hitInfo.transform.gameObject;
+            if (hitInfo.transform.gameObject.name == "Board"
+                || (_selectedPiece != null && gameObject.name.Contains("Light") != _isWhite))
             {
                 var boardPos = new Vector3(Mathf.RoundToInt(hitInfo.point.x), 0, Mathf.RoundToInt(hitInfo.point.z));
                 MakeMove(_selectedPiece, boardPos);
             }
-            else
+            else if (gameObject.name.Contains("Light") == _isWhite)
```

Comment:
Could we add a member variable to the gameObject instances of the pieces so that we don't need to do a string comparison on the name of the gameObject? An enum or a boolean flag should be good. Maybe we can set it while we load it in `SyncBoard`?